launch/launch/launch_introspector.py

```diff
@@ -111,24 +114,26 @@ def format_event_handler(event_handler: EventHandler) -> List[Text]:
 def format_action(action: Action) -> List[Text]:
     """Return a text representation of an action."""
     if is_a(action, LogInfo):
-        return ['LogInfo({})'.format(format_substitutions(action.msg))]
+        return ['LogInfo({})'.format(format_substitutions(cast(LogInfo, action).msg))]
     elif is_a(action, EmitEvent):
-        return ["EmitEvent(event='{}')".format(action.event.name)]
+        return ["EmitEvent(event='{}')".format(cast(EmitEvent, action).event.name)]
     elif is_a(action, ExecuteProcess):
+        typed_action = cast(ExecuteProcess, action)
         msg = 'ExecuteProcess(cmd=[{}], cwd={}, env={}, shell={})'.format(
-            ', '.join([format_substitutions(x) for x in action.cmd]),
-            action.cwd if action.cwd is None else "'{}'".format(
-                format_substitutions(action.cwd)
+            ', '.join([format_substitutions(x) for x in typed_action.cmd]),
+            typed_action.cwd if typed_action.cwd is None else "'{}'".format(
+                format_substitutions(typed_action.cwd)
             ),
-            action.env if action.env is None else '{' + ', '.join(
+            typed_action.env if typed_action.env is None else '{' + ', '.join(
                 ['{}: {}'.format(format_substitutions(k), format_substitutions(v))
-                 for k, v in action.env.items()] + '}'),
-            action.shell,
+                 for k, v in typed_action.env.items()] + '}'),
+            typed_action.shell,
         )
         return [msg]
     elif is_a(action, RegisterEventHandler):
-        result = ["RegisterEventHandler('{}'):".format(action.event_handler)]
-        result.extend(indent(format_event_handler(action.event_handler)))
+        typed_action2 = cast(RegisterEventHandler, action)
```

Comment:
mypy wasn't happy with reuse of the same variable name as above (even though they're in different branches)...